apps/deploy_monitoring/src/monitors/deploy.rs

```diff
@@ -0,0 +1,240 @@
+        }
+        let timestamp = Utc::now().time().to_string();
+        let file_name = format!("{}_crash.log", &timestamp);
+        let zip_name = format!("{}_crash.zip", &timestamp);
+        let file_path = format!("/tmp/tezedge-monitoring-logs/{}", &zip_name);
+
+        let mut zip_file = ZipWriter::new(File::create(&file_path)?);
+        let zip_options =
+            zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Stored);
+
+        zip_file.start_file(&file_name, zip_options)?;
+
+        // enclose the arrays in an another array to make the file a valid json
+        zip_file.write_all("[".as_bytes())?;
+
+        // last log, get its cursor_id
+        let mut cursor_id: Option<usize> =
+            match reqwest::get("http://localhost:17732/v2/log?limit=1").await {
+                Ok(result) => {
+                    let res_json = result.json::<serde_json::Value>().await?;
+                    res_json[0]["id"].as_u64().map(|val| val as usize)
+                }
+                Err(e) => {
+                    bail!("Error collecting last: {:?}", e);
+                }
+            };
+
+        // "move" trough all the possible cursor_ids with a LIMIT step
+        while let Some(cursor) = cursor_id {
+            match reqwest::get(&format!(
+                "http://localhost:17732/v2/log?node_name=9732&limit={}&cursor_id={}",
+                LIMIT, cursor
+            ))
+            .await
+            {
+                Ok(result) => {
+                    zip_file.write_all(&result.bytes().await?)?;
+                    zip_file.write_all(",".as_bytes())?;
+                }
+                Err(e) => bail!("Error collecting node logs from debugger: {:?}", e),
+            }
+            cursor_id = cursor.checked_sub(LIMIT);
+        }
+
+        zip_file.write_all("]".as_bytes())?;
+
+        let log_file_name = format!("{}/tezedge.log", TEZEDGE_VOLUME_PATH);
+        let log_file_path = Path::new(&log_file_name);
+        if log_file_path.exists() {
+            zip_file.start_file("tezedge.log", zip_options)?;
+            let file = File::open(log_file_path)?;
+            let reader = BufReader::new(file);
+
+            for line in reader.lines() {
+                zip_file.write_all(line?.as_bytes())?;
+                zip_file.write_all(&[b'\n'])?;
+            }
+        }
+
+        zip_file.finish()?;
+        Ok(file_path)
+    }
+
+    pub async fn monitor_stack(&self) -> Result<(), failure::Error> {
+        let DeployMonitor {
+            slack,
+            log,
+            compose_file_path,
+            ..
+        } = self;
+
+        if self.is_node_container_running().await {
+            let node_updated = self.changed::<TezedgeNode>().await?;
+            let debugger_updated = self.changed::<TezedgeDebugger>().await?;
+            let explorer_updated = self.changed::<Explorer>().await?;
+            // TODO: here restart everything,
```

Comment:
@adonagy 
add please leftover jira TODO here

Reply:
@adonagy Ado, what about this TODO?
remove it or add here TE-499 and add TODO here https://viablesystems.atlassian.net/browse/TE-499